### ISSUE TITLE
RR #8bb88cd: Metadata "off-by-one" & missing metadata is back, lots of `Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))` errors

### ISSUE THREAD

**Issue by Moonbase59** [AzuraCast/AzuraCast#6930]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Rolling Release Channel

### Current AzuraCast Version

Rolling Release #8bb88cd (2024-02-08 5:06)

### What happened?

Just updated to Rolling Release #8bb88cd (2024-02-08 5:06).

The "metadata missing" and "metadata off-by-one" errors are back, plus a lot of `Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))` in the Liquidsoap log:

```
2024/02/08 09:06:35 [lang:3] API nextsong - Sending POST request to 'http://127.0.0.1:6010/api/internal/1/liquidsoap/nextsong' with body: 
2024/02/08 09:06:35 [lang:3] API nextsong - Response (200): annotate:title="Kribbeln im Bauch",artist="Pe Werner",duration="227.00",song_id="92a6c5701ac9ef43137de779af9a6631",media_id="270537",playlist_id="247":media:Tagged/Werner, Pe/Werner, Pe - [compilations]/Werner, Pe - Kribbeln im Bauch.mp3
2024/02/08 09:06:35 [decoder.ffmpeg:3] Requested content-type for "/var/azuracast/stations/niteradio/media/Tagged/Werner, Pe/Werner, Pe - [compilations]/Werner, Pe - Kribbeln im Bauch.mp3": {audio=pcm(stereo)}
2024/02/08 09:06:35 [decoder.ffmpeg:3] FFmpeg recognizes "/var/azuracast/stations/niteradio/media/Tagged/Werner, Pe/Werner, Pe - [compilations]/Werner, Pe - Kribbeln im Bauch.mp3" as audio: {codec: mp3, 44100Hz, 2 channel(s)}
2024/02/08 09:06:35 [decoder.ffmpeg:3] Decoded content-type for "/var/azuracast/stations/niteradio/media/Tagged/Werner, Pe/Werner, Pe - [compilations]/Werner, Pe - Kribbeln im Bauch.mp3": {audio=pcm(stereo)}
2024/02/08 09:06:35 [cross:3] Analysis: -25.291740dB / -23.581858dB (2.98s / 3.02s)
2024/02/08 09:06:35 [cross:3] Computing crossfade duration over overlapping 2.98s buffered data at start and end.
2024/02/08 09:06:35 [lang:3] Using custom crossfade
2024/02/08 09:06:35 [cross.smart:3] No transition: using default.
2024/02/08 09:06:35 [cross_before.3:3] Content type is {audio=pcm(stereo)}.
2024/02/08 09:06:35 [buffer.7:3] Content type is {audio=pcm(stereo)}.
2024/02/08 09:06:35 [cross_after.3:3] Content type is {audio=pcm(stereo)}.
2024/02/08 09:06:35 [buffer.9:3] Content type is {audio=pcm(stereo)}.
2024/02/08 09:06:35 [cross_after_tail.2:3] Content type is {audio=pcm(stereo)}.
2024/02/08 09:06:35 [nowplaying:3] Now playing: Bay City Rollers - Summerlove Sensation
2024/02/08 09:06:35 [json.parse:3] You are parsing a JSON value without type annotation. This has confusing side-effects. Consider adding a type annotation: `let json.parse (x : ...) = ...`
2024/02/08 09:06:35 [nowplaying:3] API playlist/247 - Playlist "Pop", requests: true, scheduled: true
2024/02/08 09:06:35 [nowplaying:3] Show: "Pop (mit Musikwünschen)", color=0x2196f3, alpha=0.5
2024/02/08 09:06:36 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:06:39 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:06:42 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:06:44 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:06:47 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:06:49 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:06:51 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:06:53 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:06:55 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:06:58 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:00 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:02 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:04 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:08 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:10 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:13 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:15 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:17 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:19 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:21 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:23 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:25 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:27 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:30 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:33 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:35 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:37 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:39 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:41 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:43 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:45 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:47 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:50 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:52 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:55 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:57 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:07:59 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:01 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:03 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:05 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:07 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:09 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:11 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:13 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:16 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:19 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:21 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:24 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:26 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:28 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:30 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:32 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:35 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:37 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:40 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:42 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:45 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:47 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:49 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
2024/02/08 09:08:52 [input.http:2] Decoding failed: Avutil.Error(Server returned 401 Unauthorized (authorization failed))
```

Test stream which should play 30s song/30s beep, 30s song/30s beep, … at https://radio.niteradio.net/public/niteradio-test (web player). This nicely shows the "metadata off-by-one" effect. You can also follow "Nite Radio Test" on this web page: https://grav1.niteradio.net/blog/winterzeit-ist-radiozeit to see the effect.

The exact same setup worked fine on Rolling Release #1e2aa03 (2024-02-04 11:46), LS 2.2.4+git@6721bb156.

**Comment by Moonbase59** [comment]
Let’s handle the issue over there: https://github.com/savonet/liquidsoap/issues/3724

**Comment by Moonbase59** [comment]
**TL;DR**:
- Metadata issue seems fixed in hotfix release Liquidsoap 2.2.4-1+git@f1b53e8be. Switched my stations to that, but let’s test a few hours before integrating this.
- `Avutil.Error` _might_ have come from my one syndication playlist that addresses a remote stream. Still investigating.

**Comment by RM-FM** [comment]
Confirm, meta data update is also working again for my `sequence` edge case.

**Comment by Moonbase59** [comment]
Good to know. So let’s say this can be closed.
